drivers/source/serial_io.c

```diff
@@ -64,16 +72,29 @@ obc_error_code_t sciPrintText(unsigned char *text, uint32_t length) {
 
     /* USER CODE BEGIN */
     // Print text to the serial port using sciSendBytes. Use the mutex to protect the SCI module.
-
+    if (UART_PRINT_REG == scilinREG)
+    {
+        if (sciLinMutex != NULL)
+        {
+            if (xSemaphoreTake(sciLinMutex, UART_MUTEX_BLOCK_TIME) == pdTRUE)
```

Comment:
When printing to the other SCI interface, we should be taking `sciMutex`. Instead of passing in `sciMutex` or `scilinMutex` here, use `mutex`. It's declared near the top of this function. `mutex` is equal to either `sciMutex` or `scilinMutex` depending on what `UART_PRINT_REG` is defined as.